src/core/search/indices.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include <absl/container/flat_hash_map.h>
+
+#include <optional>
+#include <set>
+#include <vector>
+
+#include "core/search/base.h"
+
+namespace dfly::search {
+
+// Index for integer fields.
+// Range bounds are queried in logarithmic time, iteration is constant.
+struct NumericIndex : public BaseIndex {
+  void Add(DocId doc, std::string_view value) override;
+
+  std::vector<DocId> Range(int64_t l, int64_t r) const;
+
+ private:
+  std::set<std::pair<uint64_t, DocId>> entries_;
```

Comment:
can you explain what `uint64_t` is ? and why it's unsigned here but signed in `Range()`?